### PR TITLE
Feature/test storage

### PR DESCRIPTION
**Pull request by muratovv**
## What is this pull request?
This pr provide extension of storage functionality for test purposes: hard insertion of blocks and vipe storage.
   
## Why do you implement it? Why do we need this pull request?
This type of store may be very useful in integration tests for organize effective testing of components interacted with storage.
  
## How to use the features provided in the pull request?
`TestStorage` is sub class of `Storage` usage very similar to it.

## Details/Features
* `TestStorage` implementation;
* tests for it;
* reworking of `Storage` initialization;
* Reworking common test functions such as `remove_all`
* Little rework ametsuchi test, move fixture in ametsuchi_mocks

### CODE REVIEW

File: irohad/ametsuchi/impl/storage_impl.hpp [hunk 1]
```diff
@@ -54,25 +83,43 @@ namespace iroha {
...
                   std::unique_ptr<pqxx::nontransaction> wsv_transaction,
                   std::unique_ptr<WsvQuery> wsv);
-      // Storage info
+
+      /**
+       * Folder with raw blocks
+       */
       const std::string block_store_dir_;
+
+      // bd info
```
Comment (grimadas) [inline irohad/ametsuchi/impl/storage_impl.hpp]: Typo: db, or database

File: irohad/ametsuchi/impl/test_storage_impl.cpp [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    nonstd::optional<model::AccountAsset> TestStorageImpl::getAccountAsset(
+        const std::string &account_id, const std::string &asset_id) {
+      return StorageImpl::getAccountAsset(account_id, asset_id);
+    }
+    nonstd::optional<std::vector<model::Peer>> TestStorageImpl::getPeers() {
+      return StorageImpl::getPeers();
+    }
+
+  }  // namespace ametsuchi
+}  // namespace iroha
```
Comment (grimadas) [inline irohad/ametsuchi/impl/test_storage_impl.cpp]: Endline

File: libs/common/files.cpp [hunk 3]
```diff
@@ -15,35 +15,29 @@
...
       free(namelist);
     }
-    if (std::remove(dump_dir.c_str())) {
-      perror("Error deleting file");
-    }
   }
-}
-
-#endif //IROHA_AMETSUCHI_TEST_COMMON_HPP
+}
```
Comment (grimadas) [inline libs/common/files.cpp]: Endline

File: libs/common/files.hpp [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_FILES_HPP
+#define IROHA_FILES_HPP
+
+/**
+ * This source file contains common related methods
```
Comment (grimadas) [inline libs/common/files.hpp]: Not a clear comment, maybe say here that this is related to file methods  ?

File: test/framework/test_block_generator.cpp [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+#include "model/commands/create_domain.hpp"
+#include "model/commands/create_asset.hpp"
+#include "model/commands/create_account.hpp"
+#include "model/commands/set_permissions.hpp"
+#include "common/types.hpp"
+
+using namespace iroha;
+using namespace iroha::model;
+
+namespace framework {
```
Comment (grimadas) [inline test/framework/test_block_generator.cpp]: I think, we will remove this file after https://github.com/hyperledger/iroha/pull/503

File: irohad/ametsuchi/impl/test_storage_impl.cpp (hunk 2 above)
Comment (muratovv) [inline irohad/ametsuchi/impl/test_storage_impl.cpp]: Fix

File: test/framework/test_block_generator.cpp (hunk 5 above)
Comment (muratovv) [inline test/framework/test_block_generator.cpp]: We may discuss it later, i guess

File: irohad/ametsuchi/impl/storage_impl.hpp [hunk 6]
```diff
@@ -54,25 +83,43 @@ namespace iroha {
...
 
+     private:
       std::unique_ptr<FlatFile> block_store_;
+
+      /**
+       * Redis connection
+       */
       std::unique_ptr<cpp_redis::redis_client> index_;
 
       std::unique_ptr<pqxx::lazyconnection> wsv_connection_;
```
Comment (lebdron) [inline irohad/ametsuchi/impl/storage_impl.hpp]: Move this below the comment

File: irohad/ametsuchi/impl/test_storage_impl.cpp [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+                      std::move(wsv_transaction),
+                      std::move(wsv)) {
+      log_ = logger::log("TestStorage");
+    }
+
+    bool TestStorageImpl::insertBlock(model::Block block) {
+      log_->info("create mutable storage");
+      auto storage = createMutableStorage();
+      auto inserted = storage->apply(block,
+                                     [](const auto &current_block,
```
Comment (lebdron) [inline irohad/ametsuchi/impl/test_storage_impl.cpp]: Move lambda to separate variable as in `ChainValidator` as a temporary solution

File: libs/common/files.hpp [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * This source file contains common methods related to files
+ */
+
+/**
+ * Remove all files inside folder
+ * @param dump_dir - target folder
+ */
+void remove_all(const std::string &dump_dir);
```
Comment (lebdron) [inline libs/common/files.hpp]: Move to `iroha` namespace

**Pull request merged**